traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/protocol/RouterNioEndpoint.java

```diff
@@ -48,18 +54,26 @@ protected void initialiseSsl() throws Exception {
         }
     }
 
+    @SuppressWarnings({"PMD.NPathComplexity", "PMD.UseStringBufferForStringAppends"})
     synchronized private void replaceSSLHosts(final Map<String, HandshakeData> sslHostsData) {
         final Set<String> aliases = sslHostsData.keySet();
-        boolean firstAlias = true;
         String lastHostName = "";
 
-        for (final String alias : aliases) {
+        final List<String> defaultAliasList = aliases.stream().filter(alias -> alias.equalsIgnoreCase(DEFAULT_ALIAS)).collect(Collectors.toList());
+        boolean hasDefault = !defaultAliasList.isEmpty();
+        LOGGER.info( (hasDefault && defaultAliasList.size() == 1) ? "Found default alias." : "No default alias, using first from list. List length is " + aliases.size());
+        if (hasDefault && sslHostsData.get(defaultAliasList.get(0)) != null) {
+            setDefaultSSLHostConfigName(sslHostsData.get(defaultAliasList.get(0)).getHostname());
+        }
+
+        for (String alias : aliases) {
             final SSLHostConfig sslHostConfig = new SSLHostConfig();
             final SSLHostConfigCertificate cert = new SSLHostConfigCertificate(sslHostConfig, SSLHostConfigCertificate.Type.RSA);
+            sslHostConfig.setHostName(sslHostsData.get(alias).getHostname());
+            alias = alias.equalsIgnoreCase(DEFAULT_ALIAS) ? sslHostsData.get(alias).getHostname() : alias;
```

Comment:
question: this line seems strange to me, but I don't really know what's going on with this code since I haven't worked in here yet. Why does `alias` need set to the default alias hostname here if this is the default alias, otherwise `alias` is kept the same?

Reply:
no longer included with recent updates